qemu/tests/bridge_qinq.py

```diff
@@ -0,0 +1,313 @@
+    process.system_output(add_advlan_cmd)
+    advlan_iface = utils_net.Interface(advlan_ifname)
+    advlan_iface.set_mac(params["advlan_mac"])
+    advlan_iface.set_ip(params["advlan_ip"])
+    advlan_iface.set_netmask(int(netmask))
+    advlan_iface.up()
+    output = process.getoutput("ifconfig %s" % advlan_ifname)
+    logging.info(output)
+
+    # Ping guest from host via 802.1ad vlan interface
+    error_context.context("Start ping test from host to %s via %s" %
+                          (L1tag_iface_ip, advlan_ifname), logging.info)
+    ping_count = int(params.get("ping_count"))
+    status, output = utils_net.ping(L1tag_iface_ip, ping_count,
+                                    interface=advlan_ifname,
+                                    timeout=float(ping_count)*1.5)
+    if status != 0:
+        test.fail("Ping returns non-zero value %s" % output)
+    package_lost = utils_test.get_loss_ratio(output)
+    if package_lost != 0:
+        test.fail("%s packeage lost when ping guest ip %s " % (package_lost,
+                                                               L1tag_iface_ip))
+
+    # Stop tcpdump and check result
+    session.cmd_output_safe("pkill tcpdump")
+    check_tcpdump_result(session, L1tag_iface,
+                         "ethertype IPv4 (0x0800)")
+    check_tcpdump_result(session, nic_name,
+                         "ethertype 802.1Q-QinQ (0x88a8)", vlan_tag="vlan 10,")
+
+    # Set IP on L2 tag on the guest interface with vid 20
+    L2tag_iface = params["L2tag_iface"]
+    L2tag_iface_ip = params["L2tag_iface_ip"]
+    session.cmd_output("ip addr add %s/%s dev %s" % (L2tag_iface_ip, netmask,
+                                                     L2tag_iface))
+    output = session.cmd_output("ifconfig %s" % L2tag_iface, timeout=120)
+    logging.info("%s" % output)
+
+    # Start tcpdump on L1tag interface, L2tag interface and first_nic in guest
+    error_context.context("Start tcpdump in %s" % params["main_vm"],
+                          logging.info)
+    L2tag_tcpdump_log = params.get("tcpdump_log") % L2tag_iface
+    L2tag_tcpdump_cmd = params.get("tcpdump_cmd") % (L2tag_iface,
+                                                     L2tag_tcpdump_log)
+    session.sendline(L1tag_tcpdump_cmd)
+    time.sleep(2)
+    session.sendline(L2tag_tcpdump_cmd)
+    time.sleep(2)
+    session.sendline(first_nic_tcpdump_cmd)
+    time.sleep(5)
+
+    # Create 802.1q vlan via 802.1ad vlan in host
+    error_context.context("Create 802.1q vlan via 802.1ad vlan %s" %
+                          advlan_ifname, logging.info)
+    qvlan_ifname = params["qvlan_name"]
+    add_qvlan_cmd = params["add_qvlan_cmd"]
+    process.system_output(add_qvlan_cmd)
+    qvlan_iface = utils_net.Interface(qvlan_ifname)
+    qvlan_iface.set_ip(params["qvlan_ip"])
+    qvlan_iface.set_netmask(int(netmask))
+    qvlan_iface.up()
+    output = process.getoutput("ifconfig %s" % qvlan_ifname)
+    logging.info("%s" % output)
```

Comment:
Modify those`logging.info`
```suggestion
    logging.info(output)
```
Other looks good to me.